python/summarize_sgfs.py

```diff
@@ -0,0 +1,295 @@
+
+    call it by other function:
+    import summarize_sgfs
+    elo_prior = 3400
+    gameResultSummary = summarize_sgfs.GameResultSummary(elo_prior)
+    gameResultSummary.add_games(input_file_dir)
+    gameResultSummary.print_game_results()
+    gameResultSummary.print_elos()
+  """
+
+  def __init__(
+      self,
+      elo_prior: float
+  ):
+    self.results = {}  # a dictionary as {(black_player_name, white_player_name):Record}
+
+    self._all_sgfs_files = set()
+    self._all_sgf_files = set()
+    self._new_sgfs_files = set()
+    self._new_sgf_files = set()
+    self._warning = False
+    self._elo_prior = elo_prior
+
+  """ Public functions """
+
+  def add_games(self, input_file_dir, search_subdir=False):
+    """add files under input_file_dir into the results and duplicate files will not be added."""
+    self._add_game_file_names(input_file_dir, search_subdir)
+    self._add_new_games_to_result_dict()
+
+  def add_a_game_file(self, input_file_name):
+    """add a new game file (sgfs or sgf) into the results and duplicate file will not be added."""
+    self._add_a_game_file_name(input_file_name)
+    self._add_new_games_to_result_dict()
+
+  def clear(self):
+    """Clear all the records and reset to empty sets and empty dictionaries"""
+    self.results = {}  # a dictionary as {(black_player_name, white_player_name):Record}
+    self._all_sgfs_files = set()
+    self._all_sgf_files = set()
+    self._new_sgfs_files = set()
+    self._new_sgf_files = set()
+    self._warning = False
+
+  def print_elos(self) -> elo.EloInfo:
+    """Print estimated Bayes Elo scores centered as Elo prior"""
+    elo_info = self._estimate_elo()
+    print(elo_info)
+    print(f"The prior is set as {self._elo_prior}, and the estimated Bayes Elo:")
+    if self._warning:
+      warnings.warn("There are handicap games or games with komi < 5.5 or komi > 7.5")
+    return elo_info
+
+  def print_game_results(self):
+    """Print game results in a matrix with item in row i and column j as the numbers of games player i beaten player
+    j. We split draw game as 0.5 to each player."""
+    print("Player information:")
+    self._print_player_info()
+    print("Game Results by Player ID:")
+    self._print_result_matrix()
+    return
+
+  def get_game_results(self) -> Dict:
+    """Return a dictionary of game results as {(black_player_name, white_player_name):Record}
+      We can retrieve results by player's name as
+      results[(black_player_name, white_player_name)].win
+      results[(black_player_name, white_player_name)].lost
+      results[(black_player_name, white_player_name)].draw
+    """
+    return self.results
+
+  """ Private functions starts here, and no more public functions below """
+
+  def _add_game_file_names(self, input_file_dir, search_subdir):
+    if not os.path.exists(input_file_dir):
+      print(f"There is no directory under name {input_file_dir}")
+      return None
```

Comment:
Here and everywhere below: in a function that doesn't return anything, the normal convention is to just `return` rather than `return None`.

`return None` is normally used in functions that are capable of sometimes returning something besides `None`.

Reply:
Changed them all and agreed. This change will make the style consistent. But it seems that they are the same in machine code level https://stackoverflow.com/a/15300733/6598108